R/build_site_transects.R

```diff
@@ -0,0 +1,209 @@
+
+  paired_sample <- rbind(sample_points_clhs, sample_points_low_cost)
+  paired_sample$id <- paste(paste(paired_sample$bgc, paste(paired_sample$slice_num, paired_sample$point_num, sep = "."), paired_sample$cid, sep = "_"), paired_sample$rotation, sep = "_")
+
+  # Create triangle around each point and randomly rotate
+
+  cli::cli_alert_success("generating site transects")
+
+  all_triangles <- sf::st_sf(sf::st_sfc()) |> sf::st_set_crs(3005)
+
+  for (i in 1:nrow(all_points)) {
+    # i = 1
+    poc <- all_points[i, ]
+
+    triangle <- .Tri_build(id = poc$id, x = sf::st_coordinates(poc)[1], y = sf::st_coordinates(poc)[2])
+    random_rotation <- stats::runif(1, min = 0, max = 360)
+    triangle <- .rotFeature(triangle, poc, random_rotation)
+    all_triangles <- rbind(all_triangles, triangle)
+  }
+
+
+  paired_triangles <- all_triangles[all_triangles$id %in% paired_sample$id, ]
+
+  cli::cli_alert_success("generating output file to be saved : {.path {out_dir}}")
+
+  ##### write Transects####################
+
+  sf::st_write(all_points, fs::path(out_dir, outname),layer = paste0(b, "_points_all"), delete_layer = TRUE, quiet = T
+  )
+
+  sf::st_write(all_triangles, fs::path(out_dir, outname),layer = paste0(b, "_transects_all"), delete_layer = TRUE, quiet = T
+  )
+
+  sf::st_write(paired_sample, file.path(out_dir, outname),layer = paste0(b, "_points"), delete_layer = TRUE, quiet = T
+  )
+
+  #### write buffer#########################
+  triangle_buff <- sf::st_buffer(all_triangles, dist = 10)
+
+  sf::st_write(triangle_buff, fs::path(out_dir, outname), layer = paste0(b, "_transects_all_buffered"), delete_layer = TRUE, quiet = T)
+
+  sf::st_write(paired_sample, fs::path(out_dir, outname), layer = paste0(b, "_points"), delete_layer = TRUE, quiet = T)
+
+  # paired_triangles
+  sf::st_write(paired_triangles, fs::path(out_dir, outname), layer = paste0(b, "_transects"), delete_layer = TRUE, quiet = T)
+
+  #### write buffer#########################
+  ptriangle_buff <- sf::st_buffer(paired_triangles, dist = 10)
+  sf::st_write(ptriangle_buff, fs::path(out_dir, outname), layer = paste0(b, "_transects_buffered"), delete_layer = TRUE, quiet = T)
+
+  # write out clhs points only
+  sf::st_write(sample_points_clhs, fs::path(out_dir, outname), layer = paste0(b, "_points_clhs"), delete_layer = TRUE, quiet = T)
+}
+
+
+.rot <- function(a) {
+  out <- matrix(c(cos(a), sin(a), -sin(a), cos(a)), 2, 2)
+  return(out)
+}
+
+
+.Tri_build <- function(id, x, y) {
+
+  tris <- LearnGeom::CreateRegularPolygon(3, c(
+    as.numeric(paste(x)),
+    as.numeric(paste(y))
+  ), 145)
+
+  MoonLineCentre <- data.frame(tris)
+  MoonLineCentre <- sf::st_as_sf(MoonLineCentre, coords = c("X", "Y"), crs = 3005)
+  MoonLineCentre <- MoonLineCentre |>
+    dplyr::mutate(id = id) |>
+    dplyr::group_by(id) |>
+    dplyr::summarise() |>
```

Comment:
The `.by` argument is nice because it avoids having to ungroup, and it avoids non-standard evaluation shenanigans

```suggestion
    dplyr::summarise(.by = "id") |>
```